File: 2018/test/day_22_test.rb
require "test_helper"
require "day_22"

class Day22Test < Minitest::Test
  def setup
    @day_22 = Day22.new
  end

  def test_part1
    assert_equal 114, @day_22.part1(510, 10, 10)
    assert_equal 10603, @day_22.part1(6084, 14, 709)
  end
end

File: 2020/test/day_22_test.exs
defmodule Day22Test do
  use ExUnit.Case

  import Day22
  import TestHelper

  describe "Day22" do
    test "part1" do
      assert part1(input("day_22_example_1")) == 306
      assert part1(input("day_22")) == 34255
    end

    test "part2" do
      assert part2(input("day_22_example_1")) == 291
      assert part2(input("day_22")) == 33369
    end
  end
end
